test/core/test/injector-mock.test.ts

```diff
@@ -67,7 +72,1130 @@ test('correctly mocks namespaced', () => {
   `)).toMatchInlineSnapshot(`
     "const { vi } = await import('vitest')
     vi.mock('../src/add', () => {})
-    const AddModule = await import('../src/add')
-    const { default: add } = AddModule"
+    const __vi_import_0__ = await import('../src/add')"
+  `)
+})
+
+test('correctly access import', () => {
+  expect(hoistSimpleCode(`
+  import { vi } from 'vitest'
+  import add from '../src/add'
+  add();
+  vi.mock('../src/add', () => {})
+  `)).toMatchInlineSnapshot(`
+    "const { vi } = await import('vitest')
+    vi.mock('../src/add', () => {})
+    const __vi_import_0__ = await import('../src/add')
+
+      
+      
+      __vi_import_0__.default();"
   `)
 })
+
+describe('transform', () => {
+  const hoistSimpleCodeWithoutMocks = (code: string) => {
+    return hoistMocks(`import {vi} from "vitest";\nvi.mock(faker);\n${code}`, '/test.js', parse)?.code.trim()
```

Comment:
using `faker` without any reference to it feels bad. Maybe you can just use a string identifier?

Reply:
Sorry. I don't understand the second one. Do you mean I should put `vi.mock` after the original code?

Reply:
I seem to understand what you mean, I've changed to string literal. this test code makes more sense